pkg/v1/tkg/client/cluster.go

```diff
@@ -182,13 +182,7 @@ func (c *TkgClient) waitForClusterCreation(regionalClusterClient clusterclient.C
 		return errors.Wrap(err, "unable to wait for cluster nodes to be available")
 	}
 
-	if _, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableEnableAutoscaler); err == nil {
-		log.Infof("Waiting for cluster autoscaler to be available...")
-		autoscalerDeploymentName := options.ClusterName + "-cluster-autoscaler"
-		if err := regionalClusterClient.WaitForAutoscalerDeployment(autoscalerDeploymentName, options.TargetNamespace); err != nil {
-			log.Warningf("Unable to wait for autoscaler deployment to be ready. reason: %v", err)
-		}
-	}
+	c.WaitForAutoscalerDeployment(regionalClusterClient, options.ClusterName, options.TargetNamespace)
```

Comment:
**praise**: Extracted method helps make the code simpler and easier to understand.